hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -1155,11 +1157,15 @@ public void setScmTopologyClient(
   }
 
   public NetworkTopology getClusterMap() {
-    InnerNode currentTree = scmTopologyClient.getClusterTree();
-    return new NetworkTopologyImpl(configuration.get(
-        ScmConfigKeys.OZONE_SCM_NETWORK_TOPOLOGY_SCHEMA_FILE,
-        ScmConfigKeys.OZONE_SCM_NETWORK_TOPOLOGY_SCHEMA_FILE_DEFAULT),
-        currentTree);
+    InnerNode fetchedTree = scmTopologyClient.getClusterTree();
+    if (!clusterTree.equals(fetchedTree)) {
```

Comment:
Every call of getClusterMap, the two InnerNodes will be compared, which is not necessary for most of time. Why not just return the NetworkTopologyImpl from scmTopologyClient, since scmTopologyClient will periodical refresh, compare and maintain the topology tree?

Reply:
This is a good idea, I will refactor these changes to `ScmTopologyClient` for utilizing the existing comparison logic avoiding the duplicate comparison + maintenance of class variables, thanks!